### PR TITLE
changes to friday message for answering queries and test

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java [hunk 1]
```diff
@@ -29,6 +30,21 @@
 @Component
 public class StatusBuilderImpl implements StatusBuilder {
 
+  public String getFridayRetroMeetingTime(int minute) {
+    int currentMinute = minute;
+    int roundedTime = currentMinute;
+    int buffer = 5;
```
Comment (meletios) [inline src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: Should this be a constant?

File: src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java [hunk 2]
```diff
@@ -29,6 +30,21 @@
...
+
+    if(currentMinute % 5 != 0) {
+      int remainder = currentMinute % 5;
+      roundedTime = currentMinute - remainder;
+    }
+
+    int endTime = 45;
```
Comment (meletios) [inline src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: Shall we pass the endTime in the function (in case we change it)?

Comment (meletios) [reply src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: Or shall we consider it as constant?

File: src/test/java/com/hedvig/botService/chat/StatusBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static junit.framework.TestCase.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StatusBuilderTest {
+
+  @Test
+  public void test() {
```
Comment (meletios) [inline src/test/java/com/hedvig/botService/chat/StatusBuilderTest.java]: Can you give a more informative name for this function?

File: src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java (hunk 2 above)
Comment (AlexSJ1990) [reply src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: good idea - have changed to pass it into the function

File: src/test/java/com/hedvig/botService/chat/StatusBuilderTest.java (hunk 3 above)
Comment (AlexSJ1990) [reply src/test/java/com/hedvig/botService/chat/StatusBuilderTest.java]:
```suggestion
  public void stringReturnedShouldReturnExpectedWaitTime() {
```

File: src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java [hunk 4]
```diff
@@ -29,6 +30,21 @@
 @Component
 public class StatusBuilderImpl implements StatusBuilder {
 
+  public String getFridayRetroMeetingTime(int minute) {
+    int currentMinute = minute;
```
[the remark this answers is not in the document]
Comment (AlexSJ1990) [reply src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: whoops no definitely not!

File: src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java [hunk 5]
```diff
@@ -5,90 +5,205 @@
...
+@Slf4j
 public class StatusBuilderImpl implements StatusBuilder {
 
-  @Override
-  public String getStatusMessage(Clock c) {
+  private static int currentYear = Calendar.getInstance().get(Calendar.YEAR);
 
-    Instant now = Instant.now(c);
-    ZonedDateTime time = now.atZone(ZoneId.of("Europe/Stockholm"));
+  private static ArrayList<LocalDate> redDays = new ArrayList<>(Arrays.asList(LocalDate.parse(currentYear + "-01-01"),
```
Comment (meletios) [inline src/main/java/com/hedvig/botService/chat/StatusBuilderImpl.java]: i think its better if we explicitly state the red days here (as an example `2018-02-17`), and use the currentYear for the summer period

**Comment by meletios** [comment]
:tada:

**Pull request merged**